### PR TITLE
support wild card searches for label & id

### LINKED ISSUE #2849: 'Wild card' is on Filter although it 's not in any properties of item
## 🐞 Bug description
'Wild card' is on Filter although it 's not in any properties of item

**Update from Ian:** the wild-card filter searches through the `label` field.  We should change it so that it does a wildcard search through the `label` *and* `id` fields - since in use there can be some very similar labels, so id is the quickest route to finding an element.

## 🔢 Steps to Reproduce
1. Play game with data wargame-l6nngxlk
2. Click on channel 'Core Mapping'
3. Add filter 'Wild Cart'
4. Notice it can return some items
5. Click on item at step4
6. Notice no item has properties 'Wild Card', seem it gets from label

## ⚙️ Expected behavior
`Wild card' is not on Filter because of it 's not in any properties of item. 

## 🖥️ Screenshots
image

## 💻 System information
 - Browser : Chrome
 - Browser Version : Version 120.0.6099.111

### PR DESCRIPTION
**Pull request by lebaphi**
Fixes #2849

### CODE REVIEW

**Comment by Tristina1788** [comment]
Checking on branch '2849-wildcard-search' , now Wildcard search for `label` and `id` as expected

https://github.com/serge-web/serge/assets/107697044/a59fb8c5-24dd-4d8f-92af-78946fe27590

**Review by IanMayo — DISMISSED** [review]
nice fix

File: client/src/Components/local/atoms/mapping-panel/index.tsx [hunk 1]
```diff
@@ -184,11 +184,12 @@ export const MappingPanel: React.FC<MappingPanelProps> = ({ onClose, features, e
         const propertyValue = get(f.properties, filterKey, '').toString().toLowerCase()
         const searchKey = selectedFiltersProps[filterKey].value.toLowerCase()
         if (filterKey === 'Wildcard' && searchKey) {
-          // search wildcard by label
+          // search wildcard by label & id
           const label = get(f.properties, 'label', '').toString().toLowerCase()
+          const id = get(f.properties, 'id', '').toString().toLowerCase()
           try {
             const rgex = new RegExp(searchKey)
```
Comment (IanMayo) [inline client/src/Components/local/atoms/mapping-panel/index.tsx:193]: Hello @lebaphi - how about defining the regex outside the `forEach` loop?

**Comment by Tristina1788** [comment]
@IanMayo  @lebaphi  As we discussed, the label should be `ID/Label` . Now the filter is still label 'WildCard'. Should we fix it on this PR or another ticket for it?

image

**Comment by Tristina1788** [comment]
@IanMayo  Now the label is changed as expected

image

**Review by IanMayo — APPROVED** [review]

**Pull request merged**